### PR TITLE
chore: setup release automation

### LINKED ISSUE #56: add publishing credentials to repo secrets
since this repo is under the node.js org - there should be a way for @nodejs/build to publish a new version.
that can be done either by adding a repository secret and a GitHub actions flow to publish a version or by adding credentials to https://github.com/nodejs-private/secrets

### PR DESCRIPTION
**Pull request by MoLow**
Fixes: https://github.com/nodejs/tap2junit/issues/56
Refs: https://github.com/nodejs/gyp-next/pull/248

### CODE REVIEW

**Comment by MoLow** [comment]
@cclauss @jbergstroem can you please add https://pypi.org/user/nodejs-foundation/ as a maintainer of https://pypi.org/project/tap2junit/?

**Comment by legendecas** [comment]
The OCID publishing needs configurations on https://pypi.org/manage/project/tap2junit/settings/publishing/. Such as:

```
Repository: [nodejs/tap2junit](https://github.com/nodejs/tap2junit)
Workflow: release-please.yml
Environment name: pypi
```

File: .github/workflows/release-please.yml [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    - name: Upload artifact signatures to GitHub Release
+      env:
+        GITHUB_TOKEN: ${{ github.token }}
+      # Upload to GitHub Release using the `gh` CLI.
+      # `dist/` contains the built packages, and the
+      # sigstore-produced signatures and certificates.
+      run: >-
+        gh release upload
+        ${{ needs.release-please.outputs.tag_name }} dist/**
+        --repo '${{ github.repository }}'
```
Comment (legendecas) [inline .github/workflows/release-please.yml:97]: nit: missing EOF.

**Comment by MoLow** [comment]
> The OIDC publishing needs configurations on [pypi.org/manage/project/tap2junit/settings/publishing](https://pypi.org/manage/project/tap2junit/settings/publishing/). Such as:
> 
> ```
> Repository: [nodejs/tap2junit](https://github.com/nodejs/tap2junit)
> Workflow: release-please.yml
> Environment name: pypi
> ```

ping @cclauss @jbergstroem - if you have concerns or objections please raise them.
other repos and tools in the nodejs org depend on this tool and we need a way of publishing it, and if you object giving the org access please let us know so we can find an alternative way to distribute this package

File: .github/workflows/release-please.yml [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+  release-please:
+    runs-on: ubuntu-latest
+    outputs:
+      release_created: ${{ steps.release.outputs.release_created }}
+      tag_name: ${{ steps.release.outputs.tag_name }}
+    permissions:
+      contents: write
+      pull-requests: write
+    steps:
+      - uses: google-github-actions/release-please-action@v4
```
Comment (aduh95) [inline .github/workflows/release-please.yml:17]: That repo has been archived and moved to
```suggestion
      - uses: googleapis/release-please-action@v4
```

File: .github/workflows/release-please.yml [hunk 3]
```diff
@@ -0,0 +1,97 @@
+on:
+  push:
+    branches:
+      - main
+
```
Comment (aduh95) [inline .github/workflows/release-please.yml:5]: If you don't specify permission, it defaults to very loose ones
```suggestion

permissions:
  contents: read

```

File: .github/workflows/release-please.yml (hunk 2 above)
Comment (aduh95) [reply .github/workflows/release-please.yml:17]: Refs: https://github.com/google-github-actions/release-please-action
Refs: https://github.com/googleapis/release-please-action/tree/f3969c04a4ec81d7a9aa4010d84ae6a7602f86a7/

**Comment by legendecas** [comment]
Thank you @cclauss for adding nodejs-foundation as a co-owner! I have setup the OCID connection and this should be ready to go when all the suggestions are addressed.

**Comment by MoLow** [comment]
@cclauss thanks for the help!

File: .github/workflows/release-please.yml [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+permissions:
+  contents: read
```
Comment (legendecas) [inline .github/workflows/release-please.yml:7]:
```suggestion
  contents: read
  actions: read
```

File: .github/workflows/release-please.yml [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+    steps:
+      - uses: googleapis/release-please-action@v4
+        id: release
+
+  build:
+    name: Build distribution
+    needs:
+      - release-please
+    if: ${{ needs.release-please.outputs.release_created }}  # only publish on release
+    runs-on: ubuntu-latest
```
Comment (legendecas) [inline .github/workflows/release-please.yml:28]: If we change the default permission, [`actions: write`](https://docs.github.com/en/rest/actions/artifacts?apiVersion=2022-11-28) needs to be granted on this action.

```suggestion
    runs-on: ubuntu-latest
    permissions
      actions: write
```

**Review by aduh95 — APPROVED** [review]
Let's try it and adapt in a follow up if something's missing :+1:

**Review by cclauss — APPROVED** [review]

**Review by legendecas — APPROVED** [review]

**Comment by MoLow** [comment]
release worked. thanks all for helping out

**Pull request merged**